### PR TITLE
[OpenBLAS] Upgrade to v0.3.24

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Comment by amontoison** [comment]
I just remarked that OpenBLAS is compiled with `-02` and not `-03`.
Is the compilation failing with `-03`?

**Comment by giordano** [comment]
>  Is the compilation failing with -03?

I think the correct question is: how do you compile with `-O3` at all?

**Comment by giordano** [comment]
https://github.com/xianyi/OpenBLAS/blob/bb2f1ec3b01d76c44017bb65cdbf03b05e644cb4/Makefile.rule#L231-L234
```makefile
# Common Optimization Flag;
# The default -O2 is enough.
# Flags for POWER8 are defined in Makefile.power. Don't modify COMMON_OPT
# COMMON_OPT = -O2
```
seems to suggest `-O2` is enough to do whatever they want to do.

**Comment by amontoison** [comment]
Ok, so let's keep the current optimization flag.
What should we do for the failing build?

**Comment by giordano** [comment]
>  What should we do for the failing build?

@gbaraldi memory sanitize is failing with
```
[01:26:31] F90-F-0004-Unable to open MODULE file iso_fortran_env.mod (sgedmd.f90: 8)
[01:26:31] F90/x86-64 Linux Flang - 1.5 2017-05-01: compilation aborted
```

**Comment by imciner2** [comment]
The fix for the msan failure seems to be to add the following during the setup phase when all the flang compiler parts are being copied into place:
```
cp ${WORKSPACE}/x86_64-linux-musl-cxx11/destdir/include/*.mod /opt/x86_64-linux-musl/include
```

**Comment by amontoison** [comment]
The following command is not working anymore: https://buildkite.com/julialang/yggdrasil/builds/5776#018b0077-442b-4481-8486-e6daba87c91f/632-13095
Did you update the version of Clang?

**Comment by giordano** [comment]
>  Did you update the version of Clang?

Yes, it's 16.0.6 now: https://github.com/JuliaPackaging/Yggdrasil/blob/b95d2aeed3674f1790cbbb0c301406e4aa925e98/0_RootFS/LLVMBootstrap%4016/build_tarballs.jl#L7

**Comment by gbaraldi** [comment]
I think the clang builtins are in a different place, with different names.

File: S/SCOTCH/build_tarballs.jl [hunk 1]
```diff
@@ -45,6 +45,9 @@ fi
 if [[ "${target}" == *linux-musl* ]]; then
     FLAGS="-lrt -D_GNU_SOURCE"
 fi
+if [[ "${target}" == *freebsd* ]]; then
```
Comment (giordano) [inline S/SCOTCH/build_tarballs.jl:48]: This looks unrelated?

Comment (amontoison) [reply S/SCOTCH/build_tarballs.jl]: I forgot to change the branch.

**Comment by giordano** [comment]
Building for `x86_64-linux-gnu-sanitize+memory` worked locally for me, what am I missing?

**Comment by giordano** [comment]
@amontoison are you going to update openblas in Julia?

**Comment by giordano** [comment]
https://github.com/JuliaLang/julia/pull/47174

**Comment by amontoison** [comment]
Thanks Mosè! https://github.com/JuliaLang/julia/pull/51660
